packages/web-components/fast-components/src/button/button.styles.ts

```diff
@@ -1,18 +1,148 @@
 import { css } from "@microsoft/fast-element";
+import {
+    disabledCursor,
+    forcedColorsStylesheetBehavior,
+} from "@microsoft/fast-foundation";
+import { SystemColors } from "@microsoft/fast-web-utilities";
 import {
     AccentButtonStyles,
+    accentFillRestBehavior,
+    accentForegroundRestBehavior,
     BaseButtonStyles,
     LightweightButtonStyles,
+    neutralFillRestBehavior,
+    neutralFillStealthRestBehavior,
     OutlineButtonStyles,
     StealthButtonStyles,
 } from "../styles/index";
 import { appearanceBehavior } from "../utilities/behaviors";
 
 export const ButtonStyles = css`
+    :host([disabled]),
+    :host([disabled]:hover),
+    :host([disabled]:active) {
+        opacity: var(--disabled-opacity);
+        background-color: ${neutralFillRestBehavior.var};
+        cursor: ${disabledCursor};
+    }
+
     ${BaseButtonStyles}
 `.withBehaviors(
-    appearanceBehavior("accent", AccentButtonStyles),
-    appearanceBehavior("lightweight", LightweightButtonStyles),
-    appearanceBehavior("outline", OutlineButtonStyles),
-    appearanceBehavior("stealth", StealthButtonStyles)
+    forcedColorsStylesheetBehavior(
+        css`
+            :host([disabled]),
+            :host([disabled]) .control {
+                forced-color-adjust: none;
+                background-color: ${SystemColors.ButtonFace};
+                border-color: ${SystemColors.GrayText};
+                color: ${SystemColors.GrayText};
+                cursor: ${disabledCursor};
+                opacity: 1;
+            }
+        `
+    ),
+    appearanceBehavior(
+        "accent",
+        css`
+            :host([appearance="accent"][disabled]),
+            :host([appearance="accent"][disabled]:hover),
+            :host([appearance="accent"][disabled]:active) {
+                background: ${accentFillRestBehavior.var};
+            }
+
+            ${AccentButtonStyles}
+        `.withBehaviors(
+            forcedColorsStylesheetBehavior(
+                css`
+                    :host([appearance="accent"][disabled]) .control,
+                    :host([appearance="accent"][disabled]) .control:hover {
+                        background: ${SystemColors.ButtonFace};
+                        border-color: ${SystemColors.GrayText};
+                        color: ${SystemColors.GrayText};
+                    }
+                `
+            )
+        )
+    ),
+    appearanceBehavior(
+        "lightweight",
+        css`
+            :host([appearance="lightweight"][disabled]),
```

Comment:
This doesn't appear to be needed since it's already declared by the default styles for :host([appearance="lightweight"]) on lines 301-304 in patterns/button.ts

Reply:
yes, this is not needed